### DISCUSSION TITLE
Does `requires` imply `after`?

### DISCUSSION THREAD

**Discussion by greyhound-003** [wbthomason/packer.nvim#1124] [Q&A]
<sup>(Note this is a re-post. GitHub auto-flagged previous discussion as possible spam - friendly platform...)</sup>

Having [read](https://github.com/wbthomason/packer.nvim#dependencies), the  [docs](https://github.com/wbthomason/packer.nvim#sequencing) and https://github.com/wbthomason/packer.nvim/issues/810 the relation between `requires` and `after` is still not clear to me.

It seems, `requires` is for plugin installation and automatic removal (if not used anymore) and `after` determines the load order.
Does that mean, in most cases I need to specify both for proper sequencing?

Example:
```
  use 'wbthomason/packer.nvim'
  use {
    'hrsh7th/nvim-cmp',
    requires= {
      'hrsh7th/cmp-nvim-lsp',
      'hrsh7th/cmp-buffer',
      'hrsh7th/cmp-path',
      'hrsh7th/cmp-cmdline',
      {
        'saadparwaiz1/cmp_luasnip',
        requires = {
          "L3MON4D3/LuaSnip",
          tag = "v<CurrentMajor>.*"
        }
      }
    },
    config = function() require("config.nvim-cmp") end
  }
```
Intuitively, it should be clear all child dependencies need to be installed and loaded each time, before `hrsh7th/nvim-cmp` is installed/loaded. I rarely have seen someone making use of `after`.
So: Do I need to add `after` expressions here as well?

**Comment by nyngwang** [comment]
FYI: I never use `after` and never encounter any problems.
